docs/types.rst

```diff
@@ -352,6 +352,24 @@ check the value ranges at runtime and a failure causes an exception.  Enums need
         }
     }
 
+.. index:: ! contract type, ! type; contract
+
+.. _contract_types:
+
+Contracts
+---------
+
+Contracts themselves are a type that you can define and call from another
+contract, for example ::
+
+  D d = new D(4);
```

Comment:
I think a better example may be needed here.

Not sure if this is better:
> `Token token = new Token();` 
> where `Token` is a contract.

Reply:
@axic works for me, it's more descriptive.